test/integration/crud/crud_api.test.ts

```diff
@@ -304,476 +303,249 @@ describe('CRUD API', function () {
+    });
 
-        // Exercise all the options
-        cursor
-          .geoNear({ geo: 1 })
-          .group({ group: 1 })
-          .limit(10)
-          .match({ match: 1 })
-          .maxTimeMS(10)
-          .out('collection')
-          .project({ project: 1 })
-          .redact({ redact: 1 })
-          .skip(1)
-          .sort({ sort: 1 })
-          .batchSize(10)
-          .unwind('name');
-
-        // Execute the command with all steps defined
-        // will fail
-        cursor.toArray(function (err) {
-          test.ok(err != null);
-          testToArray();
-        });
-      };
-
-      //
-      // Exercise toArray
-      // -------------------------------------------------
-      const testToArray = function () {
-        const cursor = db.collection('t1').aggregate();
-        cursor.match({ a: 1 });
-        cursor.toArray(function (err, docs) {
-          expect(err).to.not.exist;
-          test.equal(3, docs.length);
-          testNext();
-        });
-      };
-
-      //
-      // Exercise next
-      // -------------------------------------------------
-      const testNext = function () {
-        const cursor = db.collection('t1').aggregate();
-        cursor.match({ a: 1 });
-        cursor.next(function (err) {
-          expect(err).to.not.exist;
-          testEach();
-        });
-      };
+    it('allMethods', async function () {
+      const cursor = db.collection('t1').aggregate([{ $match: {} }], {
+        allowDiskUse: true,
+        batchSize: 2,
+        maxTimeMS: 50
+      });
 
-      //
-      // Exercise each
-      // -------------------------------------------------
-      const testEach = function () {
-        let count = 0;
-        const cursor = db.collection('t1').aggregate();
-        cursor.match({ a: 1 });
-        cursor.forEach(
-          () => {
-            count = count + 1;
-          },
-          err => {
-            expect(err).to.not.exist;
-            test.equal(3, count);
-            testStream();
-          }
-        );
-      };
+      // Exercise all the options
+      cursor
+        .geoNear({ geo: 1 })
+        .group({ group: 1 })
+        .limit(10)
+        .match({ match: 1 })
+        .maxTimeMS(10)
+        .out('collection')
+        .project({ project: 1 })
+        .redact({ redact: 1 })
+        .skip(1)
+        .sort({ sort: 1 })
+        .batchSize(10)
+        .unwind('name');
 
-      //
-      // Exercise stream
-      // -------------------------------------------------
-      const testStream = function () {
-        const cursor = db.collection('t1').aggregate();
-        let count = 0;
-        cursor.match({ a: 1 });
-        const stream = cursor.stream();
-        stream.on('data', function () {
-          count = count + 1;
-        });
+      // Execute the command with all steps defined
+      // will fail
+      const err = await cursor.toArray().catch(err => err);
+      expect(err).to.be.instanceof(MongoServerError);
```

Comment:
Since we now catch error with `const e = await p.catch(e => e);` we get `e` initialized in both cases (resolve and reject), so I changed this assertion to `instanceof` instead of previous (`test.ok(err != null);`) which worked well with callbacks.